File: java/src/main/java/org/sparky/parser/SparkyLexerFixed.java
package org.sparky.parser;

import org.antlr.v4.runtime.CharStream;
import org.antlr.v4.runtime.Token;

/**
 * Use this not the SparkyLexer as it contains the key overrides to make the Dentation work correctly
 */
public class SparkyLexerFixed extends SparkyLexer {
    public SparkyLexerFixed(CharStream input) {
        super(input);
    }

    private final SparkyIndentationHelper denter = SparkyIndentationHelper.builder()
            .nl(NL)
            .indent(SparkyParser.INDENT)
            .dedent(SparkyParser.DEDENT)
            .pullToken(SparkyLexerFixed.super::nextToken);

    @Override
    public Token nextToken() {
        return denter.nextToken();
    }
}

File: net/Sparky/Parser/SparkyLexerFixed.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Antlr4.Runtime;

namespace Sparky.Parser
{
    public class SparkyLexerFixed : SparkyLexer
    {
        private SparkyIndentationHelper denter;

        public SparkyLexerFixed(ICharStream input)
		    : base(input)
	    {
            denter = SparkyIndentationHelper.builder()
              .nl(NL)
              .indent(SparkyParser.INDENT)
              .dedent(SparkyParser.DEDENT)
              .pullToken(base.NextToken);
        }

        public override IToken NextToken()
        {
            return denter.NextToken();
        }
    }
}
